application/aws/aws.go

```diff
@@ -169,44 +173,98 @@ func SendEmail(to, from, subject, body string) error {
 	tbody, err := html2text.FromString(body)
 	if err != nil {
 		domain.Logger.Printf("error converting html email to plain text ... %s", err.Error())
 		tbody = body
 	}
 
-	input := &ses.SendEmailInput{
-		Destination: &ses.Destination{
-			ToAddresses: []*string{
-				aws.String(to),
-			},
-		},
-		Message: &ses.Message{
-			Body: &ses.Body{
-				Html: &ses.Content{
-					Charset: aws.String("UTF-8"),
-					Data:    aws.String(body),
-				},
-				Text: &ses.Content{
-					Charset: aws.String("UTF-8"),
-					Data:    aws.String(tbody),
-				},
-			},
-			Subject: &ses.Content{
-				Charset: aws.String("UTF-8"),
-				Data:    aws.String(subject),
-			},
-		},
-		Source: aws.String(from),
-	}
-
-	result, err := svc.SendEmail(input)
+	b := &bytes.Buffer{}
+
+	b.WriteString("From: " + from + "\n")
+	b.WriteString("To: " + to + "\n")
+	b.WriteString("Subject: " + subject + "\n")
+	b.WriteString("MIME-Version: 1.0\n")
+
+	alternativeWriter := multipart.NewWriter(b)
+	b.WriteString(`Content-Type: multipart/alternative; boundary="` + alternativeWriter.Boundary() + `"` + "\n\n")
+
+	w, err := alternativeWriter.CreatePart(textproto.MIMEHeader{
+		"Content-Type":        {"text/plain", "charset=utf-8"},
+		"Content-Disposition": {"inline"},
+	})
 	if err != nil {
-		return fmt.Errorf("SendEmail failed using SES, %s", err)
+		domain.ErrLogger.Printf("failed to create MIME text part, %s", err)
+	} else {
+		_, _ = fmt.Fprint(w, tbody)
 	}
 
-	domain.Logger.Printf("Message sent using SES, message ID: %s", *result.MessageId)
-	return nil
+	relatedWriter := multipart.NewWriter(b)
+	_, err = alternativeWriter.CreatePart(textproto.MIMEHeader{
+		"Content-Type": {`multipart/related; boundary="` + relatedWriter.Boundary() + `"`},
+	})
+	if err != nil {
+		domain.ErrLogger.Printf("failed to create MIME related part, %s", err)
+	}
+
+	w, err = relatedWriter.CreatePart(textproto.MIMEHeader{
+		"Content-Type":        {"text/html", "charset=utf-8"},
+		"Content-Disposition": {"inline"},
+	})
+	if err != nil {
+		domain.ErrLogger.Printf("failed to create MIME html part, %s", err)
+	} else {
+		_, _ = fmt.Fprint(w, body)
+	}
+
+	w, err = relatedWriter.CreatePart(textproto.MIMEHeader{
+		"Content-Type":              {"image/png"},
+		"Content-Disposition":       {"inline"},
+		"Content-ID":                {"<logo>"},
+		"Content-Transfer-Encoding": {"base64"},
+	})
+	if err != nil {
+		domain.ErrLogger.Printf("failed to create MIME image part, %s", err)
+	} else {
+		logo, err := domain.Assets.Find("logo.png")
+		if err != nil {
+			domain.ErrLogger.Printf("failed to read logo file, %s", err)
+		}
+		encoder := base64.NewEncoder(base64.StdEncoding, b)
+		_, err = encoder.Write(logo)
+		if err != nil {
+			domain.ErrLogger.Printf("failed to write logo to email, %s", err)
+		}
+		err = encoder.Close()
+		if err != nil {
+			domain.ErrLogger.Printf("failed to close logo base64 encoder, %s", err)
+		}
+	}
+
+	if err = relatedWriter.Close(); err != nil {
```

Comment:
I'm wondering if it would be better to put these in a `defer` and then maybe you could put a `return` in the `if` block and not have an `else` block.